presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/ValidateDependenciesChecker.java

```diff
@@ -486,6 +487,14 @@ public Void visitRemoteSource(RemoteSourceNode node, Set<Symbol> boundSymbols)
         }
 
         @Override
+        public Void visitMerge(MergeNode node, Set<Symbol> boundSymbols)
+        {
+            verifyUniqueId(node);
+
```

Comment:
You can't, because a `MergeNode` at this point has been detached from its sources (like `RemoteSourceNode`) since it's through the `PlanFragmenter`.